### PR TITLE
back-port: Null pointer check added for system package (#1455)

### PR DESCRIPTION
**Pull request by gireeshpunathil**
Null pointer check added for system package

Hashtable does not store null values , so if systempackage comes as null it will throw null pointer exception

/cc @MohananRahul

### CODE REVIEW

**Comment by merks** [comment]
Looking here, I see there needs to be a version increment for the project being changed.

https://ci.eclipse.org/pde/job/eclipse.pde/job/PR-1459/1/console

I.e., if the containing project is version 1.0.200, it should be incremented to 1.0.201, i.e., +1 on the micro/service version.   Perhaps including features will also need such increments...

**Comment by gireeshpunathil** [comment]
@merks - thanks. looking at the CI log, the version upgradation is resolved now but new failure - not sure if it is related to infra. I am not sure if maintenance branch is well integrated into the build system

```
19:48:37  Exception in thread "Thread-3" java.lang.NoClassDefFoundError: org/eclipse/jgit/internal/JGitText
19:48:37  	at org.eclipse.jgit.internal.util.ShutdownHook.cleanup(ShutdownHook.java:85)
19:48:37  	at java.base/java.lang.Thread.run(Thread.java:840)
19:48:37  Caused by: java.lang.ClassNotFoundException: org.eclipse.jgit.internal.JGitText
19:48:37  	... 2 more
[Pipeline] }
```

**Comment by merks** [comment]
That jgit error is a very old one that is a bug in jgit and has no real build impact other than being annoying.

This looks like a "real error" where probably a version increment is needed:
```
<pre><code>MavenProject: org.eclipse.pde:org.eclipse.e4.tools.persistence.feature:1.1.300-SNAPSHOT @ /home/jenkins/agent/workspace/eclipse.pde_PR-1459/features/org.eclipse.e4.tools.persistence.feature/.polyglot.feature.xml: baseline and build artifacts have same version but different contents</code></pre>
```

Maybe @laeubi  or @HannesWell  will be better at "reading the tea leaves" here...

**Comment by HannesWell** [comment]
> I get `remote: error: GH006: Protected branch update failed for refs/heads/R4_31_maintenance.` when I attempted to land this ( `git push upstream R4_31_maintenance` ). I don't want to use the GH UI button from an experience earlier where it did not record the committer name in the commit.

Using `Rebase and merge` is perfectly fine. Only `Squash-and-merge` caused trouble, but that's now disabled for PDE.

File: features/org.eclipse.e4.tools.persistence.feature/feature.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <feature
       id="org.eclipse.e4.tools.persistence.feature"
       label="%featureName"
-      version="1.1.300.qualifier"
+      version="1.1.301.qualifier"
```
Comment (HannesWell) [inline features/org.eclipse.e4.tools.persistence.feature/feature.xml:5]: Is this version-bump necessary?

File: ui/org.eclipse.pde/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -2,6 +2,6 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %pluginName
 Bundle-SymbolicName: org.eclipse.pde;singleton:=true
-Bundle-Version: 3.13.2600.qualifier
+Bundle-Version: 3.13.2601.qualifier
```
Comment (HannesWell) [inline ui/org.eclipse.pde/META-INF/MANIFEST.MF:5]: Is this version-bump necessary?

File: features/org.eclipse.e4.tools.persistence.feature/feature.xml (hunk 1 above)
Comment (gireeshpunathil) [reply features/org.eclipse.e4.tools.persistence.feature/feature.xml:5]: as per https://ci.eclipse.org/pde/job/eclipse.pde/job/PR-1459/4/console - though it is a warning

File: ui/org.eclipse.pde/META-INF/MANIFEST.MF (hunk 2 above)
Comment (gireeshpunathil) [reply ui/org.eclipse.pde/META-INF/MANIFEST.MF:5]: I thought  I saw an error for this, but couldn't see it now! will revert.

Comment (gireeshpunathil) [reply ui/org.eclipse.pde/META-INF/MANIFEST.MF]: @HannesWell - done, thanks!

**Pull request merged**